modular_skyrat/modules/modular_items/lewd_items/code/lewd_items/shibari.dm

```diff
@@ -0,0 +1,331 @@
+//Defining rope tightness for code readability. This var works as multiplier for arousal and pleasure per tick when character tied up with those.
+#define ROPE_TIGHTNESS_LOW (1<<0)
+#define ROPE_TIGHTNESS_MED (1<<1)
+#define ROPE_TIGHTNESS_HIGH (1<<2)
+
+/obj/item/stack/shibari_rope
+	icon = 'modular_skyrat/modules/modular_items/lewd_items/icons/obj/lewd_items/lewd_items.dmi'
+	icon_state = "shibari_rope"
+	name = "shibari ropes"
+	desc = "Coil of bondage ropes."
+	amount = 1
+	merge_type = /obj/item/stack/shibari_rope
+	singular_name = "rope"
+	max_amount = 5
+	flags_1 = IS_PLAYER_COLORABLE_1
+
+	greyscale_config = /datum/greyscale_config/shibari_rope
+	greyscale_colors = "#bd8fcf"
+
+	greyscale_config_inhand_left = /datum/greyscale_config/shibari_rope_inhand_left
+	greyscale_config_inhand_right = /datum/greyscale_config/shibari_rope_inhand_right
+
+	///We use this var to change tightness var on worn version of this item.
+	var/tightness = ROPE_TIGHTNESS_LOW
+	///should clothing items created by this stack glow
+	var/glow = FALSE
+
+	///Things this rope can transform into when it's tied to a person
+	var/obj/item/clothing/under/shibari/torso/shibari_body
+	var/obj/item/clothing/under/shibari/groin/shibari_groin
+	var/obj/item/clothing/under/shibari/full/shibari_fullbody
+	var/obj/item/clothing/shoes/shibari_legs/shibari_legs
+	var/obj/item/clothing/gloves/shibari_hands/shibari_hands
+
+//This part of code spawns ropes with full stack.
+/obj/item/stack/shibari_rope/full
+	amount = 5
+
+/obj/item/stack/shibari_rope/glow
+	merge_type = /obj/item/stack/shibari_rope/glow
```

Comment:
Missing a name var to differentiate them from regular ropes. They need to differentiate themselves for the vendor, etc.